src/Stratis.Bitcoin/Mining/IBlockBuilder.cs

```diff
@@ -0,0 +1,15 @@
+﻿using NBitcoin;
+
+namespace Stratis.Bitcoin.Mining
+{
+    public interface IBlockBuilder
+    {
+        BlockTemplate Build(BlockBuilderMode mode, ChainedHeader chainTip, Script script);
+    }
+
+    public enum BlockBuilderMode
```

Comment:
Is it reasonable to move this to its own file?  Files are cheap (I know it doesn't break the policy to have multiple things per file).